### PR TITLE
feat(argo-cd): Add appHardResyncPeriod option for controller

### PR DESCRIPTION
**Pull request by jmeridth**
Base on this PR upstream https://github.com/argoproj/argo-cd/pull/8928 we can
now provide a hard resync value to the ArgoCD Application Controller

Signed-off-by: jmeridth <jmeridth@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/main/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

**Comment by jmeridth** [comment]
@pasha-codefresh do you have any concerns about this change.  I ask since you did the PR https://github.com/argoproj/argo-cd/pull/8928.  Thank you in advance if you take a look.  Cheers.

**Comment by jmeridth** [comment]
`Error: unknown flag: --app-hard-resync` looks like that this hasn't been released with argo-cd yet.  Whoops.  This will be included in [argo-cd 2.4.0](https://github.com/argoproj/argo-cd/releases/tag/v2.4.0-rc1)

**Comment by pdrastil** [comment]
This will be available via `timeout.hard.reconciliation` parameter in https://github.com/argoproj/argo-helm/pull/1267 that exposes all currently known parameters / arguments. Linking just for reference so this is not merged into PR by mistake.

**Comment by pasha-codefresh** [comment]
@jmeridth it looks good to me

**Comment by jmeridth** [comment]
@pasha-codefresh rebased onto main.  Ready to go.

I'm also looking at @pdrastil's PR #1267.  Don't want to create a regression here.  @pdrastil you ok with this possibly merging and removing from your PR as you're going with `timeout.hard.reconciliation`?  I see you're trying to break that PR apart currently.

**Comment by pdrastil** [comment]
@jmeridth I'm ok with this PR but when mine will go through this method will be deprecated and later removed. Cobra framework always process command line arguments before parsing the env vars.

**Comment by jmeridth** [comment]
@yu-croco @pdrastil @pasha-codefresh can I please snag some 👀 on this one?  @pdrastil asking again, is this worth it if #1267 is close to merge?  I don't want to add unnecessary work for that PR if this is just going to be deprecated.  Thanks.

**Comment by pdrastil** [comment]
@jmeridth - I will be splitting refactoring / argo-cmd-params today as all minor features / enhancements are in.

**Comment by jmeridth** [comment]
@pdrastil once I see those PRs, I'm comfortable closing this one.  @pasha-codefresh are you okay with this.  I know this PR complements a change you made to argo-cd.

**Comment by pdrastil** [comment]
@jmeridth You can start looking at https://github.com/argoproj/argo-helm/pull/1267. I've removed all the refactoring and did my best to keep it backwards compatible.

**Comment by jmeridth** [comment]
@pdrastil so based on [this](https://github.com/argoproj/argo-helm/pull/1267/files#diff-62394e06d563a2e131a911ece273b9320b4f71c9817fc6e406e2fc872c6c4df9R7-R9) my PR is still "relevant" as this hard resync can be a DEPRECATED option for a bit.  I see you're including [config.params.timeout.hard.reconcilliation](https://github.com/argoproj/argo-helm/pull/1267/files#diff-9b2b8960b2091a28395a78f3f4b76c4310b39062f6602f9a0575a27ed890f904R379) for future usage.

I'm willing to close this PR and then create a PR for the deprecated option once yours merges.  Or we can merge this and you can add the deprecated option for hard resync to your PR?  🤷

**Comment by pdrastil** [comment]
@jmeridth or we can merge this and I can make it deprecated when the configs.params is merged. Up to you - I'm fine with both ways.

**Comment by jmeridth** [comment]
@yu-croco @mkilchhofer could you please review this?  once reviewed we can merge and then @pdrastil can add this flag to his PR (#1267) as a deprecated option.  Thank you in advance.

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**
